telegram/ext/dispatcher.py

```diff
@@ -337,37 +337,45 @@ def process_update(self, update):
                                 try:
                                     self.persistence.update_chat_data(chat_id,
                                                                       self.chat_data[chat_id])
-                                except Exception:
-                                    self.logger.exception('Saving chat data raised an error')
+                                except Exception as e:
+                                    try:
+                                        self.dispatch_error(update, e)
+                                    except Exception:
+                                        message = 'Saving chat data raised an error and an ' \
+                                                  'uncaught error was raised while handling ' \
+                                                  'the error with an error_handler'
+                                        self.logger.exception(message)
                             if self.persistence.store_user_data and update.effective_user:
                                 user_id = update.effective_user.id
                                 try:
                                     self.persistence.update_user_data(user_id,
                                                                       self.user_data[user_id])
-                                except Exception:
-                                    self.logger.exception('Saving user data raised an error')
+                                except Exception as e:
+                                    try:
+                                        self.dispatch_error(update, e)
+                                    except Exception:
+                                        message = 'Saving user data raised an error and an ' \
+                                                  'uncaught error was raised while handling ' \
+                                                  'the error with an error_handler'
+                                        self.logger.exception(message)
                         break
 
             # Stop processing with any other handler.
             except DispatcherHandlerStop:
                 self.logger.debug('Stopping further handlers due to DispatcherHandlerStop')
                 break
 
-            # Dispatch any error.
-            except TelegramError as te:
-                self.logger.warning('A TelegramError was raised while processing the Update')
-
+            # Dispatch any error - errors should not stop the thread.
+            except Exception as e:
                 try:
-                    self.dispatch_error(update, te)
+                    self.dispatch_error(update, e)
                 except DispatcherHandlerStop:
-                    self.logger.debug('Error handler stopped further handlers')
+                    self.logger.exception('An uncaught error was raised while handling the error')
```

Comment:
Why did you change this? `DispatcherHandlerStop` is just used to prevent handlers in higher groups to be executed and is no actual exception.

Reply:
correct